# Peer-graded Assignment: Capstone Project - The Battle of Neighborhoods (Week 2)
#### Review criteria

1. A full report consisting of all of the following components:


- Introduction where you discuss the business problem and who would be interested in this project.
- Data where you describe the data that will be used to solve the problem and the source of the data.
- Methodology section which represents the main component of the report where you discuss and describe any exploratory data analysis that you did, any inferential statistical testing that you performed, if any, and what machine learnings were used and why.
- Results section where you discuss the results.
- Discussion section where you discuss any observations you noted and any recommendations you can make based on the results.
- Conclusion section where you conclude the report.


2. A link to your Notebook on your Github repository pushed showing your code.



3. Your choice of a presentation or blogpost.

## Title: Starting up successful restaurant businessin New York City

## 1. Introduction 

##### 1-1. Why New York City (NYC) ?

New York City (NYC) is the most populous city in the United States as well as worldwide. We often call United States as the "melting pot". Among big cities in States, especially New York City can be nicely expressed with the nick name above ("melting pot"). With an estimated 2018 population of 8,398,748 distributed over about 302.6 square miles (784 km**2), New York is also the most densely populated major city in the United States[1]. Located at the southern tip of the U.S. state of New York, the city is the center of the New York metropolitan area, the largest metropolitan area in the world by urban landmass.[2] With almost 20 million people in its metropolitan statistical area and approximately 23 million in its combined statistical area, it is one of the world's most populous megacities. New York City has been described as the cultural, financial, and media capital of the world, significantly influencing commerce,[3,4] entertainment, research, technology, education, politics, tourism, art, fashion, and sports. Home to the headquarters of the United Nations,[5] In addition, New York is an important center for international diplomacy. Through all of powerful environmental conditions like aboves, New York City provides various business oppourtunities and tempts to start up the business.

##### 1-2. How to approach to business item ?

In this chapter, I am going to expalain how I've approach to the target business (restaurant business) and supporting sentences for my choice. As you know, there are the neccessities of life. They are food, clothing, and shelters. First of all, I made a focus on those essential things for human life. Next, I thought about the way to extremely take benefits of the environmental characterstics of New York City. The main key of my decesion is "floating population" of New York City. As I made a comment above, population of New York City is more than 23 million. Then, let us think about floating population. New York shares its borders with five states: Pennsylvania, New Jersey, Vermont, Connecticut, and Massachusetts [6]. Numerous people are commuting to or through New York City from surrounding States/Cities. Moreover, we should remember that various people from other countries visit New York City through airway for traveling or business trip. Finally, I sorted the best item among human necessities to make fit for this floating population. Fianlly, I choose the food among human essentials and easly approach to business item as a restaurant one.    

##### 1-3. Why restaurant business ?


- The business is flexible

- The demand is high

- Different products to suite your style

- Different business structure as options

- The business is lucrative

## 2.Business problem / Approaching methodology

##### 2-1. How to choose a type of restaurant business ?

Somebody says that "New York City is made by immigrants". I understand this sentence as New York City has deep immigrant history and is multi-cultural city. The New York Times descirbes that how immigrants made the City [7]. The multi-cultural characteristics of New York City provides us to broad range of categories of foods. I carried out researches and summarize the best restaurant of New York City with food categories as below (information from business insider [8])



In [3]:
import pandas as pd
my_series = pd.Series({"AMERICAN":"Kingsley", "Malaysian":"Asiate", "GERMAN/SWISS":"Wallsé", "CANADIAN":"Hometown Bar-B-Que", "CARIBBEAN":"Ali's Roti", 
                       "PHILIPPINEAN":"PIO PIO", "DUTCH":"Black Iron Burger", "CHINESE":"Decoy", "AUSTRIAN":"Russ & Daughters", "Indonesian":"Pacificana", "French":"Le Bernardin", 
                       "Belgian":"Wallflower", "GREEK":"Milos", "INDIAN": "Indian Accent", "ITALIAN": "Marea", "JAPANESE":"Sushi Yasuda", "ITALIAN":"Marea", 
                       "KOREAN":"Jungsik", "SPANISH":"PearlOyster Bar", "Portuguese":"Meadowsweet", "MEXICAN":"Los Tacos No.1", "IRANIAN":"Tanoreen", "BRAZILIAN":"Llama Inn",
                       "THAI":"SriPraPhai", "TURKISH":"Taci's Beyti", "VIETNAMESE":"Bricolage"})
my_series

AMERICAN                  Kingsley
Malaysian                   Asiate
GERMAN/SWISS                Wallsé
CANADIAN        Hometown Bar-B-Que
CARIBBEAN               Ali's Roti
PHILIPPINEAN               PIO PIO
DUTCH            Black Iron Burger
CHINESE                      Decoy
AUSTRIAN          Russ & Daughters
Indonesian              Pacificana
French                Le Bernardin
Belgian                 Wallflower
GREEK                        Milos
INDIAN               Indian Accent
ITALIAN                      Marea
JAPANESE              Sushi Yasuda
KOREAN                     Jungsik
SPANISH            PearlOyster Bar
Portuguese             Meadowsweet
MEXICAN             Los Tacos No.1
IRANIAN                   Tanoreen
BRAZILIAN                Llama Inn
THAI                    SriPraPhai
TURKISH               Taci's Beyti
VIETNAMESE               Bricolage
dtype: object

As you can see the table above, it is truly important to make solid strategy for choosing food categories. When we start to build a strategy, we should consider at least some points as below  

- Cost (delivery, rent, parking)
- Availability to get food ingredient
- Population
- Demographics
- Neighbors
- Concept of site
- Menu of the competitors
- etc

##### 2-2. Preparation of Data Set

#### - Target site:
New York City

#### - Regal data:

- To open a restaurant in NYC, you need certain certificates, permits and licenses from various city and state agencies, including the Health Department.

- To see which type of permit and licenses you may need to apply for, visit our food service establishment permits page.

- For help opening your business, we need to visit NYC Small Business Services Start a Business portal.

- To organize and maintain all of your NYC permits in one place, sign up for NYC Business.

- Applying for a Health Department Permit/Designing a Restaurant

- Apply for a New Food Service Establishment Permit

- Suggested Guide to Food Establishment Design

We can get the information through 
https://www1.nyc.gov/site/doh/business/food-operators/opening-a-restaurant.page

#### - Boroughs and neighborgood

Neighborhood has a total of 5 boroughs and 306 neighborhoods. In order to segement the neighborhoods and explore them, we will essentially need a dataset that contains the 5 boroughs and the neighborhoods that exist in each borough as well as the the latitude and logitude coordinates of each neighborhood.
https://geo.nyu.edu/catalog/nyu_2451_34572

#### - Essential background data

- Population/floating population of New York City

- New York City demographics

- Transition type of people in New York City

- Cuisines of New York city

- Trend of the food business

We should consider some important and vital items like above. We can get the data through wikipedia and some websites as belows.

https://nypost.com/2019/12/17/the-10-worst-nyc-food-trends-of-2019

https://nypost.com/tag/food-trends

https://en.wikipedia.org/wiki/New_York_City

https://en.wikipedia.org/wiki/Economy_of_New_York_City

https://en.wikipedia.org/wiki/Portal:New_York_City

https://en.wikipedia.org/wiki/Cuisine_of_New_York_City

https://en.wikipedia.org/wiki/List_of_Michelin_starred_restaurants_in_New_York_City

#### - Geographical coordinates data set for utilizing Foursquare API

New York city geographical coordinates data will be utilized as input for the Foursquare API, that will be leveraged to provision venues information for each neighborhood.We will use the Foursquare API to explore neighborhoods in New York City. 

## 3. Explore New York City

In this section, we are going to explore New York City firstly with a geometric information. I am going to convert addresses into their equivalent latitude and longitude values. 
Next, I have a plan to try the Foursquare API to explore neighborhoods in New York City. I will use the **explore** function to get the most common venue categories in each neighborhood, and then use this feature to group the neighborhoods into clusters. 
Then, I will show you use of the *k*-means clustering algorithm to finish up stages of segmenting and clustering. Finally, I will use the Folium library to visualize the neighborhoods in New York City and their emerging clusters.

- Libarary imports:
Before we get the data and start exploring it, let's download all the dependencies that we will need.

In [2]:
import numpy as np # library to handle data in a vectorized manner

import pandas as pd # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json # library to handle JSON files

!conda install -c conda-forge geopy --yes # uncomment this line if you haven't completed the Foursquare API lab
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

!conda install -c conda-forge folium=0.5.0 --yes # uncomment this line if you haven't completed the Foursquare API lab
import folium # map rendering library

print('Libraries imported.')

Solving environment: done

# All requested packages already installed.

Solving environment: done

# All requested packages already installed.

Libraries imported.


##### 3-1. Download and Explore Dataset

Neighborhood has a total of 5 boroughs and 306 neighborhoods. In order to segement the neighborhoods and explore them, we will essentially need a dataset that contains the 5 boroughs and the neighborhoods that exist in each borough as well as the the latitude and logitude coordinates of each neighborhood. Here is the link to the dataset: https://geo.nyu.edu/catalog/nyu_2451_34572

I downloaded the files and placed it on the server, so I can simply run a `wget` command and access the data. So let's go ahead and do that.

In [4]:
!conda install -c conda-forge wget --yes

Solving environment: done

# All requested packages already installed.



In [5]:
!wget -q -O 'newyork_data.json' https://cocl.us/new_york_dataset
print('Data downloaded!')

Data downloaded!


##### 3-1-1. Load and explore the data

Next, let's load the data.

In [6]:
with open('newyork_data.json') as json_data:
    newyork_data = json.load(json_data)

Let us check that data.

In [7]:
newyork_data

{'type': 'FeatureCollection',
 'totalFeatures': 306,
 'features': [{'type': 'Feature',
   'id': 'nyu_2451_34572.1',
   'geometry': {'type': 'Point',
    'coordinates': [-73.84720052054902, 40.89470517661]},
   'geometry_name': 'geom',
   'properties': {'name': 'Wakefield',
    'stacked': 1,
    'annoline1': 'Wakefield',
    'annoline2': None,
    'annoline3': None,
    'annoangle': 0.0,
    'borough': 'Bronx',
    'bbox': [-73.84720052054902,
     40.89470517661,
     -73.84720052054902,
     40.89470517661]}},
  {'type': 'Feature',
   'id': 'nyu_2451_34572.2',
   'geometry': {'type': 'Point',
    'coordinates': [-73.82993910812398, 40.87429419303012]},
   'geometry_name': 'geom',
   'properties': {'name': 'Co-op City',
    'stacked': 2,
    'annoline1': 'Co-op',
    'annoline2': 'City',
    'annoline3': None,
    'annoangle': 0.0,
    'borough': 'Bronx',
    'bbox': [-73.82993910812398,
     40.87429419303012,
     -73.82993910812398,
     40.87429419303012]}},
  {'type': 'Feature',
 

Notice how all the relevant data is in the *features* key, which is basically a list of the neighborhoods. So, let's define a new variable that includes this data.

In [8]:
neighborhoods_data = newyork_data['features']

Let's take a look at the first item in this list.

In [9]:
neighborhoods_data[0]

{'type': 'Feature',
 'id': 'nyu_2451_34572.1',
 'geometry': {'type': 'Point',
  'coordinates': [-73.84720052054902, 40.89470517661]},
 'geometry_name': 'geom',
 'properties': {'name': 'Wakefield',
  'stacked': 1,
  'annoline1': 'Wakefield',
  'annoline2': None,
  'annoline3': None,
  'annoangle': 0.0,
  'borough': 'Bronx',
  'bbox': [-73.84720052054902,
   40.89470517661,
   -73.84720052054902,
   40.89470517661]}}

##### 3-1-2. Tranform the data into a *pandas* dataframe


The next task is essentially transforming this data of nested Python dictionaries into a *pandas* dataframe. So let's start by creating an empty dataframe.

In [10]:
# define the dataframe columns
column_names = ['Borough', 'Neighborhood', 'Latitude', 'Longitude'] 

# instantiate the dataframe
neighborhoods = pd.DataFrame(columns=column_names)

Take a look at the empty dataframe to confirm that the columns are as intended.

In [11]:
neighborhoods

Empty DataFrame
Columns: [Borough, Neighborhood, Latitude, Longitude]
Index: []

Then let's loop through the data and fill the dataframe one row at a time.

In [12]:
for data in neighborhoods_data:
    borough = neighborhood_name = data['properties']['borough'] 
    neighborhood_name = data['properties']['name']
        
    neighborhood_latlon = data['geometry']['coordinates']
    neighborhood_lat = neighborhood_latlon[1]
    neighborhood_lon = neighborhood_latlon[0]
    
    neighborhoods = neighborhoods.append({'Borough': borough,
                                          'Neighborhood': neighborhood_name,
                                          'Latitude': neighborhood_lat,
                                          'Longitude': neighborhood_lon}, ignore_index=True)

Quickly examine the resulting dataframe.

In [13]:
neighborhoods.head()

Borough Neighborhood   Latitude  Longitude
0   Bronx    Wakefield  40.894705 -73.847201
1   Bronx   Co-op City  40.874294 -73.829939
2   Bronx  Eastchester  40.887556 -73.827806
3   Bronx    Fieldston  40.895437 -73.905643
4   Bronx    Riverdale  40.890834 -73.912585

And make sure that the dataset has all 5 boroughs and 306 neighborhoods.

In [14]:
print('The dataframe has {} boroughs and {} neighborhoods.'.format(
        len(neighborhoods['Borough'].unique()),
        neighborhoods.shape[0]
    )
)

The dataframe has 5 boroughs and 306 neighborhoods.


##### 3-1-3. Use geopy library to get the latitude and longitude values of New York City.

In order to define an instance of the geocoder, we need to define a user_agent. We will name our agent <em>ny_explorer</em>, as shown below.

In [15]:
address = 'New York City, NY'

geolocator = Nominatim(user_agent="ny_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of New York City are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of New York City are 40.7127281, -74.0060152.


##### 3-1-4. Create a map of New York with neighborhoods superimposed on top and narrow down target area.


In [16]:
# create map of New York using latitude and longitude values
map_newyork = folium.Map(location=[latitude, longitude], zoom_start=10)

# add markers to map
for lat, lng, borough, neighborhood in zip(neighborhoods['Latitude'], neighborhoods['Longitude'], neighborhoods['Borough'], neighborhoods['Neighborhood']):
    label = '{}, {}'.format(neighborhood, borough)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_newyork)  
    
map_newyork

However, I referred to the information data for dynamic population of New York City. I decided to narrow down the area to Manhattan for starting up the restaurant business. As following to report[9], Simply put, Manhattan consists of much more than its residential population and daily workforce. This island, measuring just 22.96 square miles, serves approximately 4 million people on a typical weekday, 2.9 million on a weekend day, and a weekday night population of 2.05 million. let's simplify the above map and segment and cluster only the neighborhoods in Manhattan. So let's slice the original dataframe and create a new dataframe of the Manhattan data.

In [17]:
manhattan_data = neighborhoods[neighborhoods['Borough'] == 'Manhattan'].reset_index(drop=True)
manhattan_data.head()

Borough        Neighborhood   Latitude  Longitude
0  Manhattan         Marble Hill  40.876551 -73.910660
1  Manhattan           Chinatown  40.715618 -73.994279
2  Manhattan  Washington Heights  40.851903 -73.936900
3  Manhattan              Inwood  40.867684 -73.921210
4  Manhattan    Hamilton Heights  40.823604 -73.949688

Let's get the geographical coordinates of Manhattan.

In [18]:
address = 'Manhattan, NY'

geolocator = Nominatim(user_agent="ny_explorer")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of Manhattan are {}, {}.'.format(latitude, longitude))

The geograpical coordinate of Manhattan are 40.7896239, -73.9598939.


As we did with all of New York City, let's visualizat Manhattan the neighborhoods in it.

In [19]:
# create map of Manhattan using latitude and longitude values
map_manhattan = folium.Map(location=[latitude, longitude], zoom_start=11)

# add markers to map
for lat, lng, label in zip(manhattan_data['Latitude'], manhattan_data['Longitude'], manhattan_data['Neighborhood']):
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_manhattan)  
    
map_manhattan

Next, we are going to start utilizing the Foursquare API to explore the neighborhoods and segment them.

##### 3-1-5. Define Foursquare Credentials and Version

In [37]:
CLIENT_ID = '0VCIZAVPDAGGZOTMGPEWMV3SRZWHREXYGTTSR1E4QIDLAR5H' # your Foursquare ID
CLIENT_SECRET = 'WV3OIYABAOAZYQH4PR53V5GYSPVQE3Z1IVDGVUD5PSXSJSV4' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: 0VCIZAVPDAGGZOTMGPEWMV3SRZWHREXYGTTSR1E4QIDLAR5H
CLIENT_SECRET:WV3OIYABAOAZYQH4PR53V5GYSPVQE3Z1IVDGVUD5PSXSJSV4


#### Let's explore the first neighborhood in our dataframe.

As following to the GPSMYCITY [10], they choose some 'must see' site seeing spots of Manhattan. The Picked items are Times Square, Museum of Modern Art, St. Patrick's Cathedral, Rockfeller Center, Grand Central Terminal, New York Public Library, Bryant Park, Empire State Building, Macy's in Herald square, Madison Square Garden. All of top ranked sites are situated in midtown. Getting hint with these facts, once more I narrowed down the business site to Midtown of Manhattan. Get the neighborhood's name in the midtown!

In [38]:
manhattan_data.loc[15, 'Neighborhood']

'Midtown'

Get the neighborhood's latitude and longitude values.

In [39]:
neighborhood_latitude = manhattan_data.loc[15, 'Latitude'] # neighborhood latitude value
neighborhood_longitude = manhattan_data.loc[15, 'Longitude'] # neighborhood longitude value

neighborhood_name = manhattan_data.loc[15, 'Neighborhood'] # neighborhood name

print('Latitude and longitude values of {} are {}, {}.'.format(neighborhood_name, 
                                                               neighborhood_latitude, 
                                                               neighborhood_longitude))

Latitude and longitude values of Midtown are 40.75469110270623, -73.98166882730304.


##### 3-1-6. Now, let's get the top 100 venues that are in Midtown Manhattan within a radius of 500 meters.

First, let's create the GET request URL. Name your URL **url**.

In [40]:
# type your answer here

LIMIT = 100 # limit of number of venues returned by Foursquare API
radius = 500 # define radius
 # create URL
url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
    CLIENT_ID, 
    CLIENT_SECRET, 
    VERSION, 
    neighborhood_latitude, 
    neighborhood_longitude, 
    radius, 
    LIMIT)
url # display URL

'https://api.foursquare.com/v2/venues/explore?&client_id=0VCIZAVPDAGGZOTMGPEWMV3SRZWHREXYGTTSR1E4QIDLAR5H&client_secret=WV3OIYABAOAZYQH4PR53V5GYSPVQE3Z1IVDGVUD5PSXSJSV4&v=20180605&ll=40.75469110270623,-73.98166882730304&radius=500&limit=100'

Send the GET request and examine the resutls

In [41]:
results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5e933447fb34b5001cfcaeee'},
 'response': {'suggestedFilters': {'header': 'Tap to show:',
   'filters': [{'name': 'Open now', 'key': 'openNow'},
    {'name': '$-$$$$', 'key': 'price'}]},
  'headerLocation': 'Midtown East',
  'headerFullLocation': 'Midtown East, New York',
  'headerLocationGranularity': 'neighborhood',
  'totalResults': 202,
  'suggestedBounds': {'ne': {'lat': 40.759191107206235,
    'lng': -73.97573940939031},
   'sw': {'lat': 40.75019109820623, 'lng': -73.98759824521576}},
  'groups': [{'type': 'Recommended Places',
    'name': 'recommended',
    'items': [{'reasons': {'count': 0,
       'items': [{'summary': 'This spot is popular',
         'type': 'general',
         'reasonName': 'globalInteractionReason'}]},
      'venue': {'id': '3fd66200f964a520d7f11ee3',
       'name': 'Bryant Park',
       'location': {'address': 'E 42nd St',
        'crossStreet': 'btwn 5th & 6th Ave',
        'lat': 40.753621,
        'lng': -73.983265,
  

From the Foursquare lab in the previous module, we know that all the information is in the items key. Before we proceed, let's borrow the get_category_type function from the Foursquare lab.

In [42]:
# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

Now we are ready to clean the json and structure it into a *pandas* dataframe.

In [43]:
venues = results['response']['groups'][0]['items']
    
nearby_venues = json_normalize(venues) # flatten JSON

# filter columns
filtered_columns = ['venue.name', 'venue.categories', 'venue.location.lat', 'venue.location.lng']
nearby_venues =nearby_venues.loc[:, filtered_columns]

# filter the category for each row
nearby_venues['venue.categories'] = nearby_venues.apply(get_category_type, axis=1)

# clean columns
nearby_venues.columns = [col.split(".")[-1] for col in nearby_venues.columns]

nearby_venues.head()

/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: FutureWarning: pandas.io.json.json_normalize is deprecated, use pandas.json_normalize instead
  This is separate from the ipykernel package so we can avoid doing imports until


name        categories        lat        lng
0                      Bryant Park              Park  40.753621 -73.983265
1  New York Public Library Terrace             Plaza  40.753017 -73.981480
2            Nat Sherman Townhouse        Smoke Shop  40.753283 -73.980358
3           Sophie's Cuban Cuisine  Cuban Restaurant  40.755927 -73.980598
4                       sweetgreen       Salad Place  40.754640 -73.983102

Let's check top 100

In [44]:
nearby_venues.head(100)

name  \
0                                         Bryant Park   
1                     New York Public Library Terrace   
2                               Nat Sherman Townhouse   
3                              Sophie's Cuban Cuisine   
4                                          sweetgreen   
5                                                CAVA   
6                             Joanna Vargas Skin Care   
7                                    Sofitel New York   
8                                    Gabriel Kreuther   
9                            Equinox East 43rd Street   
10                                                COS   
11                                          NBA Store   
12                      Harvard Club of New York City   
13                                 Whole Foods Market   
14                                        Strip House   
15                                     Bluestone Lane   
16                                 Xi’an Famous Foods   
17                                               MUJI   
18                                     Swerve Fitness   
19              Andaz 5th Avenue - a concept by Hyatt   
20                                       Málà Project   
21                                   Books Kinokuniya   
22                    DiSO's Italian Sandwich Society   
23                               Park Italian Gourmet   
24                     The Bar Downstairs and Kitchen   
25                                              Gotan   
26                                    Belasco Theatre   
27                                       Koi New York   
28                                             adidas   
29                                      Maison Kayser   
30                                 Blue Bottle Coffee   
31                        Le Carrousel in Bryant Park   
32                                       Sunrise Mart   
33                                     Beyond Bespoke   
34                                   Desigual 530 5th   
35             Alterations.com Sewing and Alterations   
36                                     Barnes & Noble   
37                         Stone Bridge Pizza & Salad   
38                                               Joju   
39                                     The North Face   
40                              SoulCycle Bryant Park   
41                               Lady M Cake Boutique   
42                                            Ease NY   
43                                               Zara   
44                                Equinox Bryant Park   
45                                    SKECHERS Retail   
46                                         Five Below   
47                                      Ikinari Steak   
48                                      Pret A Manger   
49                                     Hudson Theatre   
50                                  db Bistro Moderne   
51                                       BCBGMAXAZRIA   
52                                   Everybody Fights   
53                                          L'ADRESSE   
54                            La Colombe Torrefaction   
55                                         Café Zaiya   
56                                            SEPHORA   
57                                        Chick-fil-A   
58                            LUMA Hotel Times Square   
59                                          TableTime   
60                                   Harvard Club Gym   
61                                            SEPHORA   
62               Pilates Reforming New York - 38th St   
63                                     Kwik Meal Cart   
64                Hyatt Centric Times Square New York   
65                             Royal Grill Halal Food   
66                                       Library Walk   
67                                            Valerie   
68                                   Burger & Lobster   
69                                     The Lambs Club   
70

And how many venues were returned by Foursquare?

In [45]:
print('{} venues were returned by Foursquare.'.format(nearby_venues.shape[0]))

100 venues were returned by Foursquare.


## 4. Explore Neighborhoods in Manhattan


#### Let's create a function to repeat the same process to all the neighborhoods in Manhattan


In [46]:
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

#### Now write the code to run the above function on each neighborhood and create a new dataframe called *manhattan_venues*.

In [47]:
manhattan_venues = getNearbyVenues(names=manhattan_data['Neighborhood'],
                                   latitudes=manhattan_data['Latitude'],
                                   longitudes=manhattan_data['Longitude']
                                  )

Marble Hill
Chinatown
Washington Heights
Inwood
Hamilton Heights
Manhattanville
Central Harlem
East Harlem
Upper East Side
Yorkville
Lenox Hill
Roosevelt Island
Upper West Side
Lincoln Square
Clinton
Midtown
Murray Hill
Chelsea
Greenwich Village
East Village
Lower East Side
Tribeca
Little Italy
Soho
West Village
Manhattan Valley
Morningside Heights
Gramercy
Battery Park City
Financial District
Carnegie Hill
Noho
Civic Center
Midtown South
Sutton Place
Turtle Bay
Tudor City
Stuyvesant Town
Flatiron
Hudson Yards


#### Let's check the size of the resulting dataframe

In [48]:
print(manhattan_venues.shape)
manhattan_venues.head()

(3176, 7)


Neighborhood  Neighborhood Latitude  Neighborhood Longitude          Venue  \
0  Marble Hill              40.876551               -73.91066       Arturo's   
1  Marble Hill              40.876551               -73.91066    Bikram Yoga   
2  Marble Hill              40.876551               -73.91066  Tibbett Diner   
3  Marble Hill              40.876551               -73.91066        Dunkin'   
4  Marble Hill              40.876551               -73.91066      Starbucks   

   Venue Latitude  Venue Longitude Venue Category  
0       40.874412       -73.910271    Pizza Place  
1       40.876844       -73.906204    Yoga Studio  
2       40.880404       -73.908937          Diner  
3       40.877136       -73.906666     Donut Shop  
4       40.877531       -73.905582    Coffee Shop

Let's check how many venues were returned for each neighborhood

In [49]:
manhattan_venues.groupby('Neighborhood').count()

Neighborhood Latitude  Neighborhood Longitude  Venue  \
Neighborhood                                                                
Battery Park City                       70                      70     70   
Carnegie Hill                           90                      90     90   
Central Harlem                          41                      41     41   
Chelsea                                100                     100    100   
Chinatown                              100                     100    100   
Civic Center                           100                     100    100   
Clinton                                100                     100    100   
East Harlem                             41                      41     41   
East Village                           100                     100    100   
Financial District                     100                     100    100   
Flatiron                               100                     100    100   
Gramercy                                84                      84     84   
Greenwich Village                      100                     100    100   
Hamilton Heights                        59                      59     59   
Hudson Yards                            70                      70     70   
Inwood                                  56                      56     56   
Lenox Hill                             100                     100    100   
Lincoln Square                          96                      96     96   
Little Italy                           100                     100    100   
Lower East Side                         55                      55     55   
Manhattan Valley                        52                      52     52   
Manhattanville                          41                      41     41   
Marble Hill                             25                      25     25   
Midtown                                100                     100    100   
Midtown South                          100                     100    100   
Morningside Heights                     38                      38     38   
Murray Hill                            100                     100    100   
Noho                                   100                     100    100   
Roosevelt Island                        25                      25     25   
Soho                                   100                     100    100   
Stuyvesant Town                         18                      18     18   
Sutton Place                           100                     100    100   
Tribeca                                 82                      82     82   
Tudor City                              78                      78     78   
Turtle Bay                             100                     100    100   
Upper East Side                         89                      89     89   
Upper West Side                         77                      77     77   
Washington Heights                      89                      89     89   
West Village                           100                     100    100   
Yorkville                              100                     100    100   

                     Venue Latitude  Venue Longitude  Venue Category  
Neighborhood                                                          
Battery Park City                70               70              70  
Carnegie Hill                    90               90              90  
Central Harlem                   41               41              41  
Chelsea                         100              100             100  
Chinatown                       100              100             100  
Civic Center                    100              100             100  
Clinton                         100              100             100  
East Harlem                      41               41              41  
East Village                    100              100             100  
Finan

#### Let's find out how many unique categories can be curated from all the returned venues


In [50]:
print('There are {} uniques categories.'.format(len(manhattan_venues['Venue Category'].unique())))

There are 329 uniques categories.


## 5. Analyze Each Neighborhood

In [51]:
# one hot encoding
manhattan_onehot = pd.get_dummies(manhattan_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
manhattan_onehot['Neighborhood'] = manhattan_venues['Neighborhood'] 

# move neighborhood column to the first column
fixed_columns = [manhattan_onehot.columns[-1]] + list(manhattan_onehot.columns[:-1])
manhattan_onehot = manhattan_onehot[fixed_columns]

manhattan_onehot.head(100)

Neighborhood  Accessories Store  Adult Boutique  Afghan Restaurant  \
0   Marble Hill                  0               0                  0   
1   Marble Hill                  0               0                  0   
2   Marble Hill                  0               0                  0   
3   Marble Hill                  0               0                  0   
4   Marble Hill                  0               0                  0   
5   Marble Hill                  0               0                  0   
6   Marble Hill                  0               0                  0   
7   Marble Hill                  0               0                  0   
8   Marble Hill                  0               0                  0   
9   Marble Hill                  0               0                  0   
10  Marble Hill                  0               0                  0   
11  Marble Hill                  0               0                  0   
12  Marble Hill                  0               0                  0   
13  Marble Hill                  0               0                  0   
14  Marble Hill                  0               0                  0   
15  Marble Hill                  0               0                  0   
16  Marble Hill                  0               0                  0   
17  Marble Hill                  0               0                  0   
18  Marble Hill                  0               0                  0   
19  Marble Hill                  0               0                  0   
20  Marble Hill                  0               0                  0   
21  Marble Hill                  0               0                  0   
22  Marble Hill                  0               0                  0   
23  Marble Hill                  0               0                  0   
24  Marble Hill                  0               0                  0   
25    Chinatown                  0               0                  0   
26    Chinatown                  0               0                  0   
27    Chinatown                  0               0                  0   
28    Chinatown                  0               0                  0   
29    Chinatown                  0               0                  0   
30    Chinatown                  0               0                  0   
31    Chinatown                  0               0                  0   
32    Chinatown                  0               0                  0   
33    Chinatown                  0               0                  0   
34    Chinatown                  0               0                  0   
35    Chinatown                  0               0                  0   
36    Chinatown                  0               0                  0   
37    Chinatown                  0               0                  0   
38    Chinatown                  0               0                  0   
39    Chinatown                  0               0                  0   
40    Chinatown                  0               0                  0   
41    Chinatown                  0               0                  0   
42    Chinatown                  0               0                  0   
43    Chinatown                  0               0                  0   
44    Chinatown                  0               0                  0   
45    Chinatown                  0               0                  0   
46    Chinatown                  0               0                  0   
47    Chinatown                  0               0                  0   
48    Chinatown                  0               0                  0   
49    Chinatown                  0               0                  0   
50    Chinatown                  0               0                  0   
51    Chinatown                  0               0                  0   
52    Chinatown                  0               0                  0   
53    Chinatown                  0               0           

And let's examine the new dataframe size.

In [52]:
manhattan_onehot.shape

(3176, 330)

#### Next, let's group rows by neighborhood and by taking the mean of the frequency of occurrence of each category


In [53]:
manhattan_grouped = manhattan_onehot.groupby('Neighborhood').mean().reset_index()
manhattan_grouped

Neighborhood  Accessories Store  Adult Boutique  Afghan Restaurant  \
0     Battery Park City           0.000000            0.00               0.00   
1         Carnegie Hill           0.000000            0.00               0.00   
2        Central Harlem           0.000000            0.00               0.00   
3               Chelsea           0.000000            0.00               0.00   
4             Chinatown           0.000000            0.00               0.00   
5          Civic Center           0.000000            0.00               0.00   
6               Clinton           0.000000            0.00               0.00   
7           East Harlem           0.000000            0.00               0.00   
8          East Village           0.000000            0.00               0.00   
9    Financial District           0.000000            0.00               0.00   
10             Flatiron           0.000000            0.00               0.00   
11             Gramercy           0.000000            0.00               0.00   
12    Greenwich Village           0.010000            0.00               0.00   
13     Hamilton Heights           0.000000            0.00               0.00   
14         Hudson Yards           0.000000            0.00               0.00   
15               Inwood           0.000000            0.00               0.00   
16           Lenox Hill           0.000000            0.00               0.01   
17       Lincoln Square           0.000000            0.00               0.00   
18         Little Italy           0.000000            0.00               0.00   
19      Lower East Side           0.000000            0.00               0.00   
20     Manhattan Valley           0.000000            0.00               0.00   
21       Manhattanville           0.000000            0.00               0.00   
22          Marble Hill           0.000000            0.00               0.00   
23              Midtown           0.000000            0.00               0.00   
24        Midtown South           0.000000            0.00               0.00   
25  Morningside Heights           0.000000            0.00               0.00   
26          Murray Hill           0.000000            0.00               0.00   
27                 Noho           0.000000            0.00               0.00   
28     Roosevelt Island           0.000000            0.00               0.00   
29                 Soho           0.000000            0.00               0.00   
30      Stuyvesant Town           0.000000            0.00               0.00   
31         Sutton Place           0.000000            0.01               0.00   
32              Tribeca           0.000000            0.00               0.00   
33           Tudor City           0.000000            0.00               0.00   
34           Turtle Bay           0.000000            0.00               0.00   
35      Upper East Side           0.000000            0.00               0.00   
36      Upper West Side           0.000000            0.00               0.00   
37   Washington Heights           0.011236            0.00               0.00   
38         West Village           0.000000            0.00               0.00   
39            Yorkville           0.000000            0.00               0.00   

    African Restaurant  American Restaurant  Antique Shop    Arcade  \
0              0.00000             0.014286          0.00  0.000000   
1              0.00000             0.011111          0.00  0.000000   
2              0.04878             0.048780          0.00  0.000000   
3              0.00000             0.040000          0.00  0.000000   
4              0.00000             0.040000          0.00  0.000000   
5              0.00000             0.030000          0.01  0.000000   
6              0.00000             0.030000          0.00  0.000000   
7              0.00000             0.000000          0.00  0.000000   
8              0.00000             0.020000       

#### Let's confirm the new size


In [54]:
manhattan_grouped.shape

(40, 330)

#### Let's print each neighborhood along with the top 5 most common venues

In [55]:
num_top_venues = 5

for hood in manhattan_grouped['Neighborhood']:
    print("----"+hood+"----")
    temp = manhattan_grouped[manhattan_grouped['Neighborhood'] == hood].T.reset_index()
    temp.columns = ['venue','freq']
    temp = temp.iloc[1:]
    temp['freq'] = temp['freq'].astype(float)
    temp = temp.round({'freq': 2})
    print(temp.sort_values('freq', ascending=False).reset_index(drop=True).head(num_top_venues))
    print('\n')

----Battery Park City----
           venue  freq
0           Park  0.09
1          Hotel  0.07
2            Gym  0.06
3  Memorial Site  0.04
4    Coffee Shop  0.04


----Carnegie Hill----
                  venue  freq
0           Coffee Shop  0.08
1                  Café  0.06
2        Cosmetics Shop  0.03
3             Bookstore  0.03
4  Gym / Fitness Center  0.03


----Central Harlem----
                 venue  freq
0    French Restaurant  0.05
1   African Restaurant  0.05
2  American Restaurant  0.05
3                  Bar  0.05
4   Chinese Restaurant  0.05


----Chelsea----
                 venue  freq
0          Coffee Shop  0.09
1          Art Gallery  0.05
2   Italian Restaurant  0.04
3  American Restaurant  0.04
4       Ice Cream Shop  0.03


----Chinatown----
                 venue  freq
0   Chinese Restaurant  0.09
1         Cocktail Bar  0.05
2  American Restaurant  0.04
3               Bakery  0.04
4         Dessert Shop  0.03


----Civic Center----
               venue  fr

As following data analysis for sorting num_top_venues, there is no restaurant among ranked places. This is positive result for us to run the restaurant in midtown. The result for midtown is as below 

----Midtown----
                 venue  freq
                 
0          Coffee Shop  0.07

1                Hotel  0.06

2       Sandwich Place  0.04

3              Theater  0.04

4  Sporting Goods Shop  0.04

#### Let's put that into a *pandas* dataframe

First, let's write a function to sort the venues in descending order.

In [56]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

Now let's create the new dataframe and display the top 10 venues for each neighborhood.

In [57]:
num_top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighborhood']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
neighborhoods_venues_sorted = pd.DataFrame(columns=columns)
neighborhoods_venues_sorted['Neighborhood'] = manhattan_grouped['Neighborhood']

for ind in np.arange(manhattan_grouped.shape[0]):
    neighborhoods_venues_sorted.iloc[ind, 1:] = return_most_common_venues(manhattan_grouped.iloc[ind, :], num_top_venues)

neighborhoods_venues_sorted.head()

Neighborhood 1st Most Common Venue 2nd Most Common Venue  \
0  Battery Park City                  Park                 Hotel   
1      Carnegie Hill           Coffee Shop                  Café   
2     Central Harlem    Seafood Restaurant                   Bar   
3            Chelsea           Coffee Shop           Art Gallery   
4          Chinatown    Chinese Restaurant          Cocktail Bar   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0                   Gym         Boat or Ferry           Coffee Shop   
1           Yoga Studio                   Gym           Pizza Place   
2    African Restaurant   American Restaurant    Chinese Restaurant   
3   American Restaurant    Italian Restaurant        Ice Cream Shop   
4                Bakery   American Restaurant                   Spa   

  6th Most Common Venue 7th Most Common Venue  8th Most Common Venue  \
0         Memorial Site             Wine Shop           Gourmet Shop   
1             Bookstore                Bakery    Japanese Restaurant   
2     French Restaurant          Dessert Shop                Library   
3                Bakery  Gym / Fitness Center              Speakeasy   
4          Dessert Shop    Salon / Barbershop  Vietnamese Restaurant   

  9th Most Common Venue 10th Most Common Venue  
0    Mexican Restaurant          Shopping Mall  
1        Cosmetics Shop     Italian Restaurant  
2  Gym / Fitness Center               Beer Bar  
3                 Hotel              Nightclub  
4     Hotpot Restaurant           Optical Shop

In [58]:
neighborhoods_venues_sorted.head(100)

Neighborhood    1st Most Common Venue    2nd Most Common Venue  \
0     Battery Park City                     Park                    Hotel   
1         Carnegie Hill              Coffee Shop                     Café   
2        Central Harlem       Seafood Restaurant                      Bar   
3               Chelsea              Coffee Shop              Art Gallery   
4             Chinatown       Chinese Restaurant             Cocktail Bar   
5          Civic Center              Coffee Shop        French Restaurant   
6               Clinton       Italian Restaurant     Gym / Fitness Center   
7           East Harlem       Mexican Restaurant                   Bakery   
8          East Village                      Bar              Pizza Place   
9    Financial District              Coffee Shop                      Bar   
10             Flatiron  New American Restaurant     Gym / Fitness Center   
11             Gramercy       Italian Restaurant              Pizza Place   
12    Greenwich Village       Italian Restaurant                     Café   
13     Hamilton Heights              Pizza Place                     Café   
14         Hudson Yards      American Restaurant                    Hotel   
15               Inwood       Mexican Restaurant                     Café   
16           Lenox Hill       Italian Restaurant              Coffee Shop   
17       Lincoln Square                    Plaza                     Café   
18         Little Italy                     Café                   Bakery   
19      Lower East Side       Chinese Restaurant                     Café   
20     Manhattan Valley              Coffee Shop       Mexican Restaurant   
21       Manhattanville              Coffee Shop       Seafood Restaurant   
22          Marble Hill           Sandwich Place                      Gym   
23              Midtown              Coffee Shop                    Hotel   
24        Midtown South        Korean Restaurant                    Hotel   
25  Morningside Heights                     Park                Bookstore   
26          Murray Hill           Sandwich Place                    Hotel   
27                 Noho       Italian Restaurant              Coffee Shop   
28     Roosevelt Island              Pizza Place                     Park   
29                 Soho              Coffee Shop       Italian Restaurant   
30      Stuyvesant Town            Boat or Ferry                     Park   
31         Sutton Place     Gym / Fitness Center       Italian Restaurant   
32              Tribeca       Italian Restaurant                     Park   
33           Tudor City                     Park                     Café   
34           Turtle Bay       Italian Restaurant              Coffee Shop   
35      Upper East Side       Italian Restaurant              Coffee Shop   
36      Upper West Side       Italian Restaurant                   Bakery   
37   Washington Heights                     Café                   Bakery   
38         West Village       Italian Restaurant  New American Restaurant   
39            Yorkville       Italian Restaurant              Coffee Shop   

   3rd Most Common Venue  4th Most Common Venue      5th Most Common Venue  \
0                    Gym          Boat or Ferry                Coffee Shop   
1            Yoga Studio                    Gym                Pizza Place   
2     African Restaurant    American Restaurant         Chinese Restaurant   
3    American Restaurant     Italian Restaurant             Ice Cream Shop   
4                 Bakery    American Restaurant                        Spa   
5                  Hotel           Cocktail Bar                Yoga Studio   
6                Theater            Coffee Shop                        Spa   
7          Deli / Bodega        Thai Restaurant  Latin American Restaurant   
8         Ice Cream Shop     Mexican Restaurant                   Wine Bar   
9            Pizza Place                  Hotel               Cocktail Ba

In the midtown, 1st to 10th most common venue is Coffee Shop, Hotel, Theater, Sandwich Place, Sporting Goods Shop, Steakhouse, Gym, Clothing, Store, Café, and Bakery. With this analysis, we can remove the steakhouse on our type of food. 


## 6. Cluster Neighborhoods

Run *k*-means to cluster the neighborhood into 5 clusters.

In [59]:
# set number of clusters
kclusters = 5

manhattan_grouped_clustering = manhattan_grouped.drop('Neighborhood', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(manhattan_grouped_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_[0:10] 

array([1, 0, 0, 0, 1, 1, 1, 3, 0, 1], dtype=int32)

Let's create a new dataframe that includes the cluster as well as the top 10 venues for each neighborhood.

In [60]:
# add clustering labels
neighborhoods_venues_sorted.insert(0, 'Cluster Labels', kmeans.labels_)

manhattan_merged = manhattan_data

# merge toronto_grouped with toronto_data to add latitude/longitude for each neighborhood
manhattan_merged = manhattan_merged.join(neighborhoods_venues_sorted.set_index('Neighborhood'), on='Neighborhood')

manhattan_merged.head() # check the last columns!

Borough        Neighborhood   Latitude  Longitude  Cluster Labels  \
0  Manhattan         Marble Hill  40.876551 -73.910660               4   
1  Manhattan           Chinatown  40.715618 -73.994279               1   
2  Manhattan  Washington Heights  40.851903 -73.936900               3   
3  Manhattan              Inwood  40.867684 -73.921210               3   
4  Manhattan    Hamilton Heights  40.823604 -73.949688               3   

  1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
0        Sandwich Place                   Gym           Coffee Shop   
1    Chinese Restaurant          Cocktail Bar                Bakery   
2                  Café                Bakery         Grocery Store   
3    Mexican Restaurant                  Café                Lounge   
4           Pizza Place                  Café           Coffee Shop   

  4th Most Common Venue 5th Most Common Venue 6th Most Common Venue  \
0           Yoga Studio         Deli / Bodega       Supplement Shop   
1   American Restaurant                   Spa          Dessert Shop   
2         Deli / Bodega    Chinese Restaurant     Mobile Phone Shop   
3           Pizza Place            Restaurant    Chinese Restaurant   
4         Deli / Bodega    Mexican Restaurant      Sushi Restaurant   

  7th Most Common Venue  8th Most Common Venue 9th Most Common Venue  \
0            Steakhouse          Shopping Mall    Seafood Restaurant   
1    Salon / Barbershop  Vietnamese Restaurant     Hotpot Restaurant   
2    Mexican Restaurant        Supplement Shop           Coffee Shop   
3                  Park                 Bakery   American Restaurant   
4          Cocktail Bar         Sandwich Place                Bakery   

      10th Most Common Venue  
0                Pizza Place  
1               Optical Shop  
2  Latin American Restaurant  
3         Frozen Yogurt Shop  
4                Yoga Studio

Finally, let's visualize the resulting clusters

In [61]:
# create map
map_clusters = folium.Map(location=[latitude, longitude], zoom_start=11)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(manhattan_merged['Latitude'], manhattan_merged['Longitude'], manhattan_merged['Neighborhood'], manhattan_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

## 7. Examine Clusters

Now, I can examine each cluster and determine the discriminating venue categories that distinguish each cluster. Based on the defining categories, I can then assign a name to each cluster. 

#### Cluster 1

In [62]:
manhattan_merged.loc[manhattan_merged['Cluster Labels'] == 0, manhattan_merged.columns[[1] + list(range(5, manhattan_merged.shape[1]))]]

Neighborhood 1st Most Common Venue    2nd Most Common Venue  \
6      Central Harlem    Seafood Restaurant                      Bar   
8     Upper East Side    Italian Restaurant              Coffee Shop   
9           Yorkville    Italian Restaurant              Coffee Shop   
10         Lenox Hill    Italian Restaurant              Coffee Shop   
12    Upper West Side    Italian Restaurant                   Bakery   
13     Lincoln Square                 Plaza                     Café   
17            Chelsea           Coffee Shop              Art Gallery   
18  Greenwich Village    Italian Restaurant                     Café   
19       East Village                   Bar              Pizza Place   
20    Lower East Side    Chinese Restaurant                     Café   
21            Tribeca    Italian Restaurant                     Park   
23               Soho           Coffee Shop       Italian Restaurant   
24       West Village    Italian Restaurant  New American Restaurant   
25   Manhattan Valley           Coffee Shop       Mexican Restaurant   
27           Gramercy    Italian Restaurant              Pizza Place   
30      Carnegie Hill           Coffee Shop                     Café   
31               Noho    Italian Restaurant              Coffee Shop   
34       Sutton Place  Gym / Fitness Center       Italian Restaurant   
35         Turtle Bay    Italian Restaurant              Coffee Shop   

   3rd Most Common Venue  4th Most Common Venue     5th Most Common Venue  \
6     African Restaurant    American Restaurant        Chinese Restaurant   
8                 Bakery   Gym / Fitness Center                   Exhibit   
9                    Gym                    Bar             Deli / Bodega   
10      Sushi Restaurant            Pizza Place              Cocktail Bar   
12                   Bar            Coffee Shop           Thai Restaurant   
13    Italian Restaurant  Performing Arts Venue                   Theater   
17   American Restaurant     Italian Restaurant            Ice Cream Shop   
18     French Restaurant       Sushi Restaurant                       Gym   
19        Ice Cream Shop     Mexican Restaurant                  Wine Bar   
20                Bakery    Japanese Restaurant               Coffee Shop   
21                  Café                    Spa                  Wine Bar   
23              Boutique         Clothing Store  Mediterranean Restaurant   
24              Wine Bar    American Restaurant                    Bakery   
25     Indian Restaurant                    Bar               Yoga Studio   
27                   Bar            Coffee Shop       American Restaurant   
30           Yoga Studio                    Gym               Pizza Place   
31           Pizza Place                  Hotel         French Restaurant   
34           Coffee Shop            Pizza Place                       Gym   
35                 Hotel                   Café          Sushi Restaurant   

   6th Most Common Venue   7th Most Common Venue 8th Most Common Venue  \
6      French Restaurant            Dessert Shop               Library   
8            Yoga Studio               Wine Shop             Juice Bar   
9            Pizza Place        Sushi Restaurant             Wine Shop   
10                  Café            Burger Joint                   Gym   
12            Shoe Store      Seafood Restaurant                  Café   
13          Concert Hall               Wine Shop   American Restaurant   
17                Bakery    Gym / Fitness Center             Speakeasy   
18          Dessert Shop             Coffee Shop    Chinese Restaurant   
19           Coffee Shop            Cocktail Bar             Speakeasy   
20                  Park        Ramen Restaurant           Art Gallery   
21           Coffee Shop     American Restaurant           Men's Store   
23                 Hotel       French Restaurant                Bakery   
24                  Park            Cocktail Bar             Jazz

#### Cluster 2

In [63]:
manhattan_merged.loc[manhattan_merged['Cluster Labels'] == 1, manhattan_merged.columns[[1] + list(range(5, manhattan_merged.shape[1]))]]

Neighborhood    1st Most Common Venue 2nd Most Common Venue  \
1            Chinatown       Chinese Restaurant          Cocktail Bar   
14             Clinton       Italian Restaurant  Gym / Fitness Center   
15             Midtown              Coffee Shop                 Hotel   
16         Murray Hill           Sandwich Place                 Hotel   
22        Little Italy                     Café                Bakery   
28   Battery Park City                     Park                 Hotel   
29  Financial District              Coffee Shop                   Bar   
32        Civic Center              Coffee Shop     French Restaurant   
33       Midtown South        Korean Restaurant                 Hotel   
38            Flatiron  New American Restaurant  Gym / Fitness Center   
39        Hudson Yards      American Restaurant                 Hotel   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
1                 Bakery   American Restaurant                   Spa   
14               Theater           Coffee Shop                   Spa   
15               Theater        Sandwich Place   Sporting Goods Shop   
16                   Bar           Coffee Shop      Sushi Restaurant   
22                 Hotel       Bubble Tea Shop        Sandwich Place   
28                   Gym         Boat or Ferry           Coffee Shop   
29           Pizza Place                 Hotel          Cocktail Bar   
32                 Hotel          Cocktail Bar           Yoga Studio   
33   Japanese Restaurant          Dessert Shop          Burger Joint   
38   Japanese Restaurant    Italian Restaurant   American Restaurant   
39  Gym / Fitness Center    Italian Restaurant               Dog Run   

     6th Most Common Venue 7th Most Common Venue     8th Most Common Venue  \
1             Dessert Shop    Salon / Barbershop     Vietnamese Restaurant   
14         Thai Restaurant          Cocktail Bar            Sandwich Place   
15              Steakhouse                   Gym            Clothing Store   
16     Japanese Restaurant           Pizza Place              Burger Joint   
22            Cocktail Bar    Chinese Restaurant  Mediterranean Restaurant   
28           Memorial Site             Wine Shop              Gourmet Shop   
29                    Park   American Restaurant        Mexican Restaurant   
32                    Park                   Spa          Sushi Restaurant   
33          Clothing Store   American Restaurant      Gym / Fitness Center   
38  Furniture / Home Store                  Café              Cycle Studio   
39      Spanish Restaurant          Burger Joint                      Café   

   9th Most Common Venue         10th Most Common Venue  
1      Hotpot Restaurant                   Optical Shop  
14                   Gym                          Hotel  
15                  Café                         Bakery  
16  Gym / Fitness Center       Mediterranean Restaurant  
22       Thai Restaurant                 Ice Cream Shop  
28    Mexican Restaurant                  Shopping Mall  
29   Japanese Restaurant           Gym / Fitness Center  
32   American Restaurant           Gym / Fitness Center  
33           Coffee Shop                    Salad Place  
38                   Spa  Vegetarian / Vegan Restaurant  
39       Thai Restaurant                            Gym

#### Cluster 3

In [64]:
manhattan_merged.loc[manhattan_merged['Cluster Labels'] == 2, manhattan_merged.columns[[1] + list(range(5, manhattan_merged.shape[1]))]]

Neighborhood 1st Most Common Venue 2nd Most Common Venue  \
37  Stuyvesant Town         Boat or Ferry                  Park   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
37                   Bar           Gas Station        Baseball Field   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
37       Harbor / Marina     German Restaurant          Cocktail Bar   

   9th Most Common Venue 10th Most Common Venue  
37           Coffee Shop               Heliport

#### Cluster 4

In [65]:
manhattan_merged.loc[manhattan_merged['Cluster Labels'] == 3, manhattan_merged.columns[[1] + list(range(5, manhattan_merged.shape[1]))]]

Neighborhood 1st Most Common Venue 2nd Most Common Venue  \
2    Washington Heights                  Café                Bakery   
3                Inwood    Mexican Restaurant                  Café   
4      Hamilton Heights           Pizza Place                  Café   
5        Manhattanville           Coffee Shop    Seafood Restaurant   
7           East Harlem    Mexican Restaurant                Bakery   
11     Roosevelt Island           Pizza Place                  Park   
26  Morningside Heights                  Park             Bookstore   
36           Tudor City                  Park                  Café   

   3rd Most Common Venue 4th Most Common Venue      5th Most Common Venue  \
2          Grocery Store         Deli / Bodega         Chinese Restaurant   
3                 Lounge           Pizza Place                 Restaurant   
4            Coffee Shop         Deli / Bodega         Mexican Restaurant   
5     Italian Restaurant                  Park         Mexican Restaurant   
7          Deli / Bodega       Thai Restaurant  Latin American Restaurant   
11                   Gym           Coffee Shop          Kosher Restaurant   
26           Coffee Shop   American Restaurant              Deli / Bodega   
36      Greek Restaurant    Mexican Restaurant              Deli / Bodega   

   6th Most Common Venue    7th Most Common Venue  8th Most Common Venue  \
2      Mobile Phone Shop       Mexican Restaurant        Supplement Shop   
3     Chinese Restaurant                     Park                 Bakery   
4       Sushi Restaurant             Cocktail Bar         Sandwich Place   
5     Chinese Restaurant       Spanish Restaurant  Check Cashing Service   
7      Convenience Store       Spanish Restaurant           Liquor Store   
11      Greek Restaurant                  Dog Run           Liquor Store   
26          Burger Joint  New American Restaurant      Outdoor Sculpture   
36                 Diner              Coffee Shop   Gym / Fitness Center   

   9th Most Common Venue     10th Most Common Venue  
2            Coffee Shop  Latin American Restaurant  
3    American Restaurant         Frozen Yogurt Shop  
4                 Bakery                Yoga Studio  
5                   Bank                        Bar  
7            Gas Station                 Taco Place  
11        Sandwich Place             Scenic Lookout  
26         Grocery Store           Greek Restaurant  
36            Restaurant                     Garden

#### Cluster 5

In [66]:
manhattan_merged.loc[manhattan_merged['Cluster Labels'] == 4, manhattan_merged.columns[[1] + list(range(5, manhattan_merged.shape[1]))]]

Neighborhood 1st Most Common Venue 2nd Most Common Venue  \
0  Marble Hill        Sandwich Place                   Gym   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0           Coffee Shop           Yoga Studio         Deli / Bodega   

  6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
0       Supplement Shop            Steakhouse         Shopping Mall   

  9th Most Common Venue 10th Most Common Venue  
0    Seafood Restaurant            Pizza Place

## 8. Result and discussion

We need to consider the cluster data. Like seen in above, our business target area is in cluster 2. I just focus on midtown south. In midtwon south, common venues are Korean Restaurants, Hotels, Japanese Restaurant, Dessert Shop, Burger Joint, Clothing Store, American Restaurant, Gym / Fitness Center, Coffee Shop, Salad Places. We might delete those kinds of restaurant like Korean, Japanese, American restaurant and burger shop.

We can get one more hint from race population ratio of New York City. According to the research Health of Latinos in New York City by the New York City Department of Health and Mental Hygiene's, NYC residents who identify themselves as Latino or Hispanic account for nearly a third of the city's population. From 2000 to 2015, the Latino population grew by more than 14% to 2,485,125.The largest heritage groups in NYC are Puerto Ricans and Dominicans. More than Half of Latinos are US-born. Of Latinos born outside of the US, nearly three-quarters have lived in the US for 10 years or more [11]. We can also refer the demographic profile of New York City [12]. Spanish and Latino hold 27.7 % of population of New York City right after 40% of whites. 

If you can remember our approaching process above, you should easily pop up we removed American food, steakhouse, burgers like foods coming from western side. As following all processes aboves, finally, I can make a dicision to start up the Spanish Restaurant on midtown of Manhattan. 

## 9. Conclusion

If you can remember our approaching process above, you should easily pop up we removed American food, steakhouse, burgers like foods coming from western side. As following all processes aboves, finally, I can make a dicision to start up the Spanish Restaurant on midtown of Manhattan. Thank you for following me! 

## References

[1] Mike Maciag (October 2, 2013). "Mapping the Nation's Most Densely Populated Cities". Governing – The States and Localities. Archived from the original on August 27, 2016. Retrieved August 27, 2016.

[2] "World Urban Areas" (PDF). Demographia. 2018. Retrieved March 27, 2018..

[3] "Top 8 Cities by GDP: China vs. The U.S." Business Insider, Inc. July 31, 2011. Retrieved July 1, 2018. For instance, Shanghai, the largest Chinese city with the highest economic production, and a fast-growing global financial hub, is far from matching or surpassing New York, the largest city in the U.S. and the economic and financial super center of the world.

[4] "New York City: The Financial Capital of the World". Pando Logic. October 8, 2015. Retrieved July 1, 2018.

[5] "Plan your visit". United Nations Visitor Centre. Archived from the original on March 14, 2017. Retrieved February 9, 2017. The Headquarters of the United Nations is located in New York City, along the East River. When you pass through the gates of the United Nations visitors' entrance, you enter an international territory. This 18-acre site does not belong to just one country, but to all countries that have joined the Organization; currently, the United Nations has 193 Member States.

[6] Which States Border New York State? - WorldAtlas.comwww.worldatlas.com › articles › which-states-border-new-york-state

[7] https://www.nytimes.com/2016/11/06/books/review/city-of-dreams-history-of-immigrant-new-york-tyler-anbinder.html

[8] https://www.businessinsider.com/best-nyc-restaurant-for-every-cuisine-2016-12

[9] The dynamic population of mahattan, Mitchell L. Moss and Carson Qing Rudin Center for Transportation Policy and Management, Wagner School of Public Service, New York University, 2019

[10] https://www.gpsmycity.com/tours/midtown-manhattan-walking-tour-1623.html

[11] https://withcanopy.com/blog/a-snapshot-of-the-latino-population-in-nyc

[12] https://en.wikipedia.org/wiki/Demographics_of_New_York_City